src/goto-analyzer/show_on_source.cpp

```diff
@@ -0,0 +1,142 @@
+/*******************************************************************\
+
+Module: goto-analyzer
+
+Author: Daniel Kroening, kroening@kroening.com
+
+\*******************************************************************/
+
+#include "show_on_source.h"
+
+#include <util/file_util.h>
+#include <util/message.h>
+#include <util/unicode.h>
+
+#include <analyses/ai.h>
+
+#include <fstream>
+
+static bool show_location(const ai_baset &ai, ai_baset::locationt loc)
```

Comment:
All callers re-do `.full_path()`, which is non-trivial. How about returning an `optionalt<std::string>` so that this can be avoided? As an added value, that would also better fit the procedure name I'd say.